dags/generate_dags.py

```diff
@@ -23,12 +25,25 @@ def generate_dags():
         key = file_["Key"]
         if key.endswith("/"):
             continue
+        file_name = Path(key).stem
         result = client.get_object(Bucket=bucket, Key=key)
-        collection = result["Body"].read().decode()
-        collection = json.loads(collection)
-        if collection.get("schedule"):
+        discovery_configs = result["Body"].read().decode()
+        discovery_configs = json.loads(discovery_configs)
+
+        # Allow the file content to be either one config or a list of configs
+        if type(discovery_configs) is dict:
+            discovery_configs = [discovery_configs]
+        scheduled_discovery_configs = [
+            discovery_config
+                for discovery_config in discovery_configs
+                    if discovery_config.get("schedule")
+            ]
+        for idx, discovery_config in enumerate(scheduled_discovery_configs):
+            id = f"discover-{file_name}"
+            if idx > 0:
+                id = f"{id}-{idx}"
             get_discover_dag(
-                id=f"discover-{collection['collection']}", event=collection
+                id=id, event=discovery_config
```

Comment:
Rather than creating a new DAG for each `event` in the file, we could pass `scheduled_collections` to `get_discover_dag`, and return something that looks like this:

```
discover_grp = subdag_discover.expand(event=scheduled_collections)
```

This works using the MWAA version increment in #159 

That way, instead of a new DAG for each `idx`, we create one dag per file input. The only issue would be coalescing the schedules in each event.